src/v0.8/core/finance/escrow/EscrowDataTradingFee.sol

```diff
@@ -24,22 +24,101 @@ import {MatchingType} from "src/v0.8/types/MatchingType.sol";
 
 import {EscrowBase} from "src/v0.8/core/finance/escrow/EscrowBase.sol";
 
+import {ArraysPaymentInfoLIB} from "src/v0.8/shared/utils/array/ArrayLIB.sol";
+
 /// @title EscrowDataTradingFee
 /// @dev This contract provides functions for managing EscrowDataTradingFee-related operations.
 contract EscrowDataTradingFee is EscrowBase {
-    /// @dev Internal function to get bids refund information.
+    using ArraysPaymentInfoLIB for FinanceType.PaymentInfo[];
+
+    /// @dev Internal function to get refund information.
+    /// @param _datasetId The ID of the dataset.
     /// @param _matchingId The ID of the matching process.
+    /// @param _payers An array containing the addresses of the dataset and matching process payers.
+    /// @param _token The type of token for escrow handling (e.g., FIL, ERC-20).
     /// @return refunds An array containing payment information for refund.
-    function _getBidsRefundInfo(
-        uint64 /*_datasetId*/,
+    function _getRefundInfo(
+        uint64 _datasetId,
         uint64 _matchingId,
-        address /*_token*/
+        address[] memory _payers,
+        address _token
     )
         internal
         view
         override
         returns (FinanceType.PaymentInfo[] memory refunds)
     {
+        refunds = new FinanceType.PaymentInfo[](0);
+
+        if (_isEscrowRefund(_matchingId)) {
+            FinanceType.PaymentInfo[] memory refundInfo = super._getRefundInfo(
+                _datasetId,
+                _matchingId,
+                _payers,
+                _token
+            );
+            refunds = refunds.appendArrays(refundInfo);
+        }
+
+        if (_isBidsRefund(_matchingId)) {
+            FinanceType.PaymentInfo[] memory refundInfo = _getBidsRefundInfo(
+                _datasetId,
+                _matchingId,
+                _token
+            );
+            refunds = refunds.appendArrays(refundInfo);
+        }
+
+        if (_isParentAccountRefund(_datasetId)) {
+            FinanceType.PaymentInfo[]
+                memory refundInfo = _getParentAccountRefundInfo(
+                    _datasetId,
+                    _token
+                );
+            refunds = refunds.appendArrays(refundInfo);
+        }
+    }
+
+    /// @dev Internal function to get parent account refund information.
+    /// @param _datasetId The ID of the dataset.
+    /// @param _token The type of token for escrow handling (e.g., FIL, ERC-20).
+    /// @return refunds An array containing payment information for refund.
+    function _getParentAccountRefundInfo(
```

Comment:
recommendation is to rename _getParentAccountRefundInfo to _getMoveSourceAccountRefundInfo